### ISSUE TITLE
borg check --repair fails to repair "Invalid segment magic"

### ISSUE THREAD

**Issue by pattop** [borgbackup/borg#4440]
Borg check started reporting invalid segment magic. Attempting a borg check --repair results in the following failure:

```
'check --repair' is an experimental feature that might result in data loss.
Type 'YES' if you understand this and want to continue: YES
Data integrity error: Invalid segment magic [segment 67453, offset 0]
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/borg/repository.py", line 911, in check
    objects = list(self.io.iter_objects(segment))
  File "/usr/lib/python3.7/site-packages/borg/repository.py", line 1335, in iter_objects
    raise IntegrityError('Invalid segment magic [segment {}, offset {}]'.format(segment, 0))
borg.helpers.IntegrityError: Data integrity error: Invalid segment magic [segment 67453, offset 0]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4436, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 4368, in run
    return set_ec(func(args))
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 152, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3.7/site-packages/borg/archiver.py", line 313, in do_check
    if not repository.check(repair=args.repair, save_space=args.save_space):
  File "/usr/lib/python3.7/site-packages/borg/repository.py", line 916, in check
    self.io.recover_segment(segment, filename)
  File "/usr/lib/python3.7/site-packages/borg/repository.py", line 1364, in recover_segment
    with mmap.mmap(backup_fd.fileno(), 0, access=mmap.ACCESS_READ) as mm:
ValueError: cannot mmap an empty file

Platform: Linux gtr 5.0.0-arch1-1-ARCH #1 SMP PREEMPT Mon Mar 4 14:11:43 UTC 2019 x86_64
Linux: arch  
Borg: 1.1.8  Python: CPython 3.7.2
PID: 20024  CWD: /smb/pcmirror
sys.argv: ['/usr/bin/borg', 'check', '--repair', '.']
SSH_ORIGINAL_COMMAND: None
```

**Comment by ThomasWaldmann** [comment]
I remember having fixed this a while ago (have a look at the changelog whether it is in 1.1.9).

It is because your segment file has 0 bytes and mmap can't deal with that.

**Comment by ThomasWaldmann** [comment]
Duplicate of #4272 and fixed in 1.1.9.

**Comment by pattop** [comment]
Thanks.
